corehq/apps/users/bulk_download.py

```diff
@@ -191,7 +192,8 @@ def parse_mobile_users(domain, user_filters, task=None, total_count=None):
             user_groups_length = max(user_groups_length, len(group_names))
             max_location_length = max(max_location_length, len(user_dict["location_code"]))
             if task:
-                DownloadBase.set_progress(task, n, total_count)
+                DownloadBase.set_progress(task, n + current_user_downloaded_count, total_count)
+        current_user_downloaded_count += n + 1
```

Comment:
I just saw [this forum post](https://forum.dimagi.com/t/downloading-users-progress-bar-showing-126/8014), which made me think of this PR.  I think this logic is incorrect - the progress is set to `n + current_user_downloaded_count`, but `current_user_downloaded_count` has already been incremented for each of those `n`s.  Plus, it should only be incremented by one - it's currently being incremented by the position of the current user in its domain.

Note that this issue will also affect single domain downloads.

```python
In [1]: current_user_downloaded_count = 0 
   ...: for current_domain in [['a', 'b'], ['c', 'd', 'e']]: 
   ...:     for n, user in enumerate(current_domain): 
   ...:         progress = n + current_user_downloaded_count 
   ...:         print(f"User: {user}  {progress}/4") 
   ...:         current_user_downloaded_count += n + 1 
   ...:                                                                                
User: a  0/4
User: b  2/4
User: c  3/4
User: d  5/4
User: e  8/4
```

I think you just want one global counter outside of the loop incremented for each user, like this:

```python
In [2]: n = 0 
   ...: for current_domain in [['a', 'b'], ['c', 'd', 'e']]: 
   ...:     for user in current_domain: 
   ...:         print(f"User: {user}  {n}/4") 
   ...:         n += 1 
   ...:                                                                                
User: a  0/4
User: b  1/4
User: c  2/4
User: d  3/4
User: e  4/4
```

Reply:
@esoergel Hi Ethan! Thanks for pointing out the forum post-- I agree it looks progress bar related. 
For your example above were by user 'e' the progress showed 8/4, the indentation  is slightly different than what's in place after this PR. Adding `n` to `current_user_downloaded_count` occurs outside the loop of enumerating through the users in the current domain.
```
>>> current_user_downloaded_count = 0 
... for current_domain in [['a', 'b'], ['c', 'd', 'e']]:
...      for n, user in enumerate(current_domain):
...              progress = n + current_user_downloaded_count
...              print(f"User: {user}  {progress}/4")
...      current_user_downloaded_count += n + 1
...
User: a  0/4
User: b  1/4
User: c  2/4
User: d  3/4
User: e  4/4
```
I think the global variable is a bit more clean, and I'm happy to make that fix, I just also want to make sure to also fix the issue from the forum post if this is not the cause.  ( Correct me if any of this logic is off)

Reply:
Oh, you're absolutely right - apologies, I'd misread this PR.  No need to make that change unless you particularly want to.  Well now I'm confused as to what the cause of that forum issue is - might be a discrepancy between users in ES and in couch.